src/libmongoc/src/mongoc/mongoc-uri.c

```diff
@@ -2503,6 +2506,30 @@ mongoc_uri_get_option_as_int32 (const mongoc_uri_t *uri,
    return (int32_t) retval;
 }
 
+
+static void
+_mongoc_uri_warn_for_bad_int_option_combos (const mongoc_uri_t *uri,
+					    const char *option)
+{
+   /* Warn for deprecated option combinations */
+   if (!strcasecmp (option, MONGOC_URI_TIMEOUTMS)) {
+      if (mongoc_uri_get_option_as_int64 (uri, MONGOC_URI_WAITQUEUETIMEOUTMS, -1) > 0 ||
+	  mongoc_uri_get_option_as_int64 (uri, MONGOC_URI_SOCKETTIMEOUTMS, -1) > 0 ||
+	  mongoc_uri_get_option_as_int64 (uri, MONGOC_URI_WTIMEOUTMS, -1) > 0) {
+	 MONGOC_WARNING ("Setting a deprecated timeout option in combination with timeoutMS");
```

Comment:
To give users more information on which option is deprecated:
```suggestion
	 MONGOC_WARNING ("Setting a deprecated timeout option %s in combination with timeoutMS", option);
```